### PR TITLE
Update `contributing.md` PR submission process [ci skip]

### PR DESCRIPTION
**Pull request by lougeniaC64**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Comment by jdragojevic** [comment]
@lougeniaC64  - At some point we'd discussed removing the references to `a-s-dev`.  That would be a good improvement since that section takes up most the visible space.

**Comment by lougeniaC64** [comment]
> @lougeniaC64 - At some point we'd discussed removing the references to `a-s-dev`. That would be a good improvement since that section takes up most the visible space.
I vaguely remember us having that discussion but I wasn't sure if we made a final decision. I'll remove that section and request review from Mark to see if he has any objections.

**Comment by lougeniaC64** [comment]
> @lougeniaC64 - At some point we'd discussed removing the references to `a-s-dev`. That would be a good improvement since that section takes up most the visible space.

I vaguely remember us having that discussion but I wasn't sure if we made a final decision. I'll remove that section and request review from Mark to see if he has any objections.

File: docs/contributing.md [hunk 1]
```diff
@@ -56,23 +33,23 @@ Below are a few different queries you can use to find appropriate issues to work
...
-- Your patch should include new tests that cover your changes, or be accompanied by explanation for why it doesn't need any. It is your
-  and your reviewer's responsibility to ensure your patch includes adequate tests.
+- Your patch should include new tests that cover your changes, or be accompanied by explanation for why it doesn't need any. It is your and your reviewer's responsibility to ensure your patch includes adequate tests.
   - Consult the [testing guide](./howtos/testing-a-rust-component.md) for some tips on writing effective tests.
+- Your code should pass all the automated tests before you submit your PR for review.
+  - The simplest way to confirm this is to run `cargo all_tests`, which uses the `./automation/all_tests.sh` script, that runs all test suites and linters for Rust, Kotlin and Swift code.
+    - **Note:** You should avoid running all automated tests to incrementally validate your changes as it can be a long-running process. Instead run a reasonable subset of all tests that will exercise your changes and then run `cargo all_tests` to validate your entire patch before submitting your PR.
```
Comment (mhammond) [inline docs/contributing.md:40]: I think the "should" here is a bit too strong - something like "you might choose to" might make it clearer that you are just suggesting an optimization.

File: docs/contributing.md [hunk 2]
```diff
@@ -56,23 +33,23 @@ Below are a few different queries you can use to find appropriate issues to work
...
+  - "Work in progress" pull requests are welcome, but should be clearly labeled as such and should not be merged until all tests pass and the code has been reviewed.
+    - You can label pull requests as "Work in progress" by creating them as draft PRs and adding "[WIP]" to the beginning of the pull request title.
```
Comment (mhammond) [inline docs/contributing.md:42]: This was probably written before github actually had first-class support for WIP - ie, I think we might as well change this to "by using the github PR UI to indicate this PR is a draft - [learn more about draft PRs](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests)"

**Review by mhammond — APPROVED** [review]
This is great, thanks.

**Pull request merged**